Sources/PitchSpeller/Data Structures/Graph.swift

```diff
@@ -29,11 +29,15 @@ public struct Graph <Value: Hashable>: Hashable {
 
         public let source: Node
         public let destination: Node
-        public var value: Double
+        public var value: EdgeCapacity
+        
+        public var hashValue: Int {
```

Comment:
I'd say except for exceptional circumstances, we needn't implement our own `hashValue`. If all of the stored types here are themselves `Hashable`, we should get `Hashable` for free. Was the compiler whining that `Graph` no longer conformed to `Hashable`? Either we can relax the requirement for `Graph` to be `Equatable`, rather than `Hashable`, but considering we get it for free…

Reply:
yeah it was complaining! I think you may have to do it explicitly when you are using enums with associated values...

Reply:
Hmmm, not if the associated value types are themselves `Hashable`.

You can try: 

``` Swift
struct AssociatedType: Hashable {
    let value: Int
}

enum A: Hashable {
    case special
    case normal(AssociatedType)
}
```

Which will compile. If you take away `Hashable` conformance from `AssociatedType`, it will no longer compiler.